deep_code/tools/publish.py

```diff
@@ -206,10 +232,18 @@ def _update_variable_catalogs(self, generator, file_dict, variable_ids):
                 )
                 file_dict[var_file_path] = updated_catalog.to_dict()
 
-    def publish_dataset(self, write_to_file: bool = False):
+    def publish_dataset(
+        self,
+        write_to_file: bool = False,
+        mode: Literal["all", "dataset", "workflow"] = "all",
+    ) -> dict[str, Any]:
         """Prepare dataset/product collection for publishing to the specified GitHub
         repository."""
 
+        if not self.dataset_config:
+            raise ValueError(
+                "No dataset config loaded. Provide dataset_config_path to publish dataset."
```

Comment:
```suggestion
                "No dataset config loaded. Provide dataset_config to publish dataset."
```

Isn't the parameter called `dataset_config`?